libsolidity/interface/CompilerStack.cpp

```diff
@@ -150,10 +150,19 @@ bool CompilerStack::parse()
 			}
 		}
 	}
-	if (!Error::containsOnlyWarnings(m_errors))
-		// errors while parsing. should stop before type checking
+	if (Error::containsOnlyWarnings(m_errors))
+	{
+		m_stackState = ParsingSuccessful;
+		return true;
+	}
+	else
 		return false;
+}
 
+bool CompilerStack::analyze()
+{
+	if (m_stackState < SourcesSet)
```

Comment:
yeah, right. i had the import function in mind, which would not need to have the ast parsed, but that should be introduced when the import function is ready,